### PR TITLE
1.5 set new password kozhukh

### PR DESCRIPTION
**Pull request by OksanaKozhukh**

### CODE REVIEW

File: front/src/common/redux/set-password/set.password.api.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import axios from 'axios';
+
+export async function setPassswordApi (request) {
+    const url = `${process.env.REACT_APP_SET_PASSWORD_ROUT}${request.data.token}`;
+    const sentData = JSON.stringify(request.data);
```
Comment (staspitsyk) [inline front/src/common/redux/set-password/set.password.api.js]: Alex told me, that this variable sentData is redundant.

File: front/src/common/redux/set-password/set.password.reducer.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+const initialState = {
+    isSetPassword: false,
+    errorMessage:''
+}
+
+export function setPassword (state = initialState, action) {
+    switch(action.type) {
+        case types.SET_PASSWORD_SUCCESS:
+            return {...state, isSetPassword: true, errorMessage:''};
```
Comment (staspitsyk) [inline front/src/common/redux/set-password/set.password.reducer.js]: errorMessage is not changing at this line, I think you should not write it, as it is already in ...state

**Review by so2niko — APPROVED** [review]
ok

**Review by maslovakat — APPROVED** [review]

File: front/src/common/redux/set-password/set.password.saga.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { put, call, all, takeEvery } from 'redux-saga/effects';
+import { setPassswordApi } from './set.password.api';
+import * as types from './types';
+
+function* setPasswordSaga(payload) {
+    try {
+        const response = yield call(setPassswordApi, payload);
+        yield put({ type: types.SET_PASSWORD_SUCCESS, response });
```
Comment (Alan18081) [inline front/src/common/redux/set-password/set.password.saga.js]: Please, use in actions only `type` and `payload` inside of actions

File: front/src/common/redux/set-password/set.password.saga.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    } catch (error) {
+        yield put({ type: types.SET_PASSWORD_ERROR, error });
+    }
+}
+
+function* watchSetPassword() {
+    yield takeEvery(types.SET_PASSWORD, setPasswordSaga);
```
Comment (Alan18081) [inline front/src/common/redux/set-password/set.password.saga.js]: Move this `takeEvery` to the `passwordSaga`

File: front/src/common/redux/set-password/set.password.saga.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import { put, call, all, takeEvery } from 'redux-saga/effects';
+import { setPassswordApi } from './set.password.api';
+import * as types from './types';
+
+function* setPasswordSaga(payload) {
+    try {
+        const response = yield call(setPassswordApi, payload);
+        yield put({ type: types.SET_PASSWORD_SUCCESS, response });
+    } catch (error) {
+        yield put({ type: types.SET_PASSWORD_ERROR, error });
```
Comment (Alan18081) [inline front/src/common/redux/set-password/set.password.saga.js]: Please, use in actions only `type` and `payload` inside of actions

File: front/src/common/redux/set-password/set.password.api.js (hunk 1 above)
Comment (OksanaKozhukh) [reply front/src/common/redux/set-password/set.password.api.js]: Fixed

File: front/src/common/redux/set-password/set.password.reducer.js (hunk 2 above)
Comment (OksanaKozhukh) [reply front/src/common/redux/set-password/set.password.reducer.js]: Fixed

File: front/src/common/redux/set-password/set.password.saga.js (hunk 3 above)
Comment (OksanaKozhukh) [reply front/src/common/redux/set-password/set.password.saga.js]: Fixed

File: front/src/common/redux/set-password/set.password.saga.js (hunk 4 above)
Comment (OksanaKozhukh) [reply front/src/common/redux/set-password/set.password.saga.js]: Fixed

File: front/src/common/redux/set-password/set.password.saga.js (hunk 5 above)
Comment (OksanaKozhukh) [reply front/src/common/redux/set-password/set.password.saga.js]: Fixed

File: front/src/common/redux/set-password/set.password.api.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import axios from 'axios';
+
+export async function setPassswordApi (request) {
+    const url = `${process.env.REACT_APP_SET_PASSWORD_ROUT}${request.data.token}`;
+    const headers = { headers: { 'Content-Type': 'application/json' } };
+   
+    return axios
+        .put(url, JSON.stringify(request.data), headers)
```
Comment (Alan18081) [inline front/src/common/redux/set-password/set.password.api.js]: We don't need `JSON.stringify` here, just pass `request.data` here

Comment (OksanaKozhukh) [reply front/src/common/redux/set-password/set.password.api.js]: Fixed

File: front/src/common/redux/set-password/set.password.api.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import axios from 'axios';
+
+export async function setPassswordApi (request) {
+
+    const url = `${process.env.REACT_APP_SET_PASSWORD_ROUT}${request.data.token}`;
+    const headers = { headers: { 'Content-Type': 'application/json' } };
+   
+    return axios
+        .put(url, request.data, headers)
+        .then(response => response)
```
Comment (Alan18081) [inline front/src/common/redux/set-password/set.password.api.js]: Please, return only data from API function, don't return the full response, saga should not know anything about HTTP requests

Comment (OksanaKozhukh) [reply front/src/common/redux/set-password/set.password.api.js]: Fixed

File: front/src/common/redux/set-password/set.password.api.js [hunk 8]
```diff
@@ -0,0 +1,11 @@
+import axios from 'axios';
+
+export async function setPassswordApi (request) {
+
+    const url = `${process.env.REACT_APP_SET_PASSWORD_ROUT}${request.data.token}`;
```
Comment (Alan18081) [inline front/src/common/redux/set-password/set.password.api.js:5]: Please, rename it to `REACT_APP_SET_PASSWORD_ROUT` to `REACT_APP_SET_PASSWORD_ROUTE`

**Review by Alan18081 — APPROVED** [review]

**Pull request merged**